pkg/machinery/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -1264,13 +1239,46 @@ func (m *InstallDiskSizeMatcher) UnmarshalYAML(unmarshal func(interface{}) error
 		return fmt.Errorf("failed to parse disk size %s: %s", parts[2], err)
 	}
 
-	m.Matcher = func(d *disk.Disk) bool {
-		return compare(d, size)
+	m.MatchData = MatchData{
+		Op:   op,
+		Size: size,
 	}
 
 	return nil
 }
 
+// Matcher is a method that can handle some custom disk matching logic.
+func (m *InstallDiskSizeMatcher) Matcher(d *disk.Disk) bool {
+	return m.MatchData.Compare(d)
+}
+
+// MatchData contains data for comparison - Op and Size.
+// docgen:nodoc
```

Comment:
@smira I'm not sure about this line, is this correct usage?

Reply:
```suggestion
//
//docgen:nodoc
```

this should be good style-wise, actually docgen is @Unix4ever baby

Reply:
Done